cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -726,6 +727,60 @@ public NamespaceMeta apply(AppMds input) throws Exception {
     });
   }
 
+
+  @Override
+  public void addAdapter(final Id.Namespace id, final AdapterSpecification adapterSpecification) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        mds.apps.writeAdapter(id, adapterSpecification);
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public AdapterSpecification getAdapter(final Id.Namespace id, final String name) throws OperationException {
+    return txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, AdapterSpecification>() {
+      @Override
+      public AdapterSpecification apply(AppMds mds) throws Exception {
+        return mds.apps.getAdapter(id, name);
+      }
+    });
+  }
+
+  @Override
+  public Collection<AdapterSpecification> getAllAdapters(final Id.Namespace id) throws OperationException {
+    return txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Collection<AdapterSpecification>>() {
+      @Override
+      public Collection<AdapterSpecification> apply(AppMds mds) throws Exception {
+        return mds.apps.getAllAdapters(id);
+      }
+    });
+  }
+
+  @Override
+  public void removeAdapter(final Id.Namespace id, final String name) throws OperationException {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        mds.apps.deleteAdapter(id, name);
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public void removeAllAdapters(final Id.Namespace id) throws OperationException {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        mds.apps.deleteAllAdapters(id);
```

Comment:
Ok, I see its the same for Application. It just seemed unusual to simply delegate to a method named `deleteApp` from a `removeApp`